docs/spire.md

```diff
@@ -0,0 +1,287 @@
+1. After the TaskRun steps complete, as part of the entrypointer code, it requests an SVID from SPIFFE workload API (via the SPIRE agent socket)
+1. The SPIRE agent will attest the workload and request an SVID.
+1. The entrypointer receives an x509 SVID, containing the x509 certificate and associated private key. 
+1. The entrypointer signs the results of the TaskRun and emits the signatures and x509 certificate to the TaskRun results for later verification.
+
+## Enabling TaskRun result attestations
+
+To enable TaskRun attestations:
+1. Make sure `enable-spire` is set to `"true"` in the `feature-flags` configmap, see [`install.md`](./install.md#customizing-the-pipelines-controller-behavior) for details
+1. Create a SPIRE deployment containing a SPIRE server, SPIRE agents and the SPIRE CSI driver, for convenience, [this sample single cluster deployment](https://github.com/spiffe/spiffe-csi/tree/main/example/config) can be used.
+1. Register the SPIRE workload entry for Tekton with the "Admin" flag, which will allow the Tekton controller to communicate with the SPIRE server to manage the TaskRun identities dynamically.
+    ```
+
+    # This example is assuming use of the above SPIRE deployment
+    # Example where trust domain is "example.org" and cluster name is "example-cluster"
+    
+    # Register a node alias for all nodes of which the Tekton Controller may reside
+    kubectl -n spire exec -it \
+        deployment/spire-server -- \
+        /opt/spire/bin/spire-server entry create \
+            -node \
+            -spiffeID spiffe://example.org/allnodes \
+            -selector k8s_psat:cluster:example-cluster
+    
+    # Register the tekton controller workload to have access to creating entries in the SPIRE server
+    kubectl -n spire exec -it \
+        deployment/spire-server -- \
+        /opt/spire/bin/spire-server entry create \
+            -admin \
+            -spiffeID spiffe://example.org/tekton/controller \
+            -parentID spiffe://example.org/allnode \
+            -selector k8s:ns:tekton-pipelines \
+            -selector k8s:pod-label:app:tekton-pipelines-controller \
+            -selector k8s:sa:tekton-pipelines-controller
+    
+    ```
+    
+1. Modify the controller (`config/controller.yaml`) to provide access to the SPIRE agent socket.
+    ```yaml
+    # Add the following the volumeMounts of the "tekton-pipelines-controller" container
+    - name: spiffe-workload-api
+      mountPath: /spiffe-workload-api
+      readOnly: true
+    
+    # Add the following to the volumes of the controller pod
+    - name: spiffe-workload-api
+      csi:
+        driver: "csi.spiffe.io"
+    ```
+1. (Optional) Modify the controller (`config/controller.yaml`) to configure non-default SPIRE options by adding arguments to the CLI.
```

Comment:
Isn't this done via config map now?

Reply:
Thats correct. Updated the doc to reflect the change.
Thanks for pointing this out.